File: meeting_service/meeting_service/schemas/meeting.py
"""Meeting template schemas for request/response validation."""

from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field

from meeting_service.core.enums import EmployeeLevel, MeetingType


class MeetingBase(BaseModel):
    """Base meeting template schema."""

    title: str = Field(..., min_length=1, max_length=200)
    description: str | None = Field(None, max_length=1000)
    type: MeetingType
    department_id: int | None = None
    position: str | None = Field(None, max_length=100)
    level: EmployeeLevel | None = None
    is_mandatory: bool = True
    order: int = 0
    deadline_days: int = Field(default=7, ge=0, le=365)
    duration_minutes: int = Field(default=60, ge=1, le=480)


class MeetingCreate(MeetingBase):
    """Meeting creation schema."""


class MeetingUpdate(BaseModel):
    """Meeting update schema (all fields optional)."""

    title: str | None = Field(None, min_length=1, max_length=200)
    description: str | None = Field(None, max_length=1000)
    type: MeetingType | None = None
    department_id: int | None = None
    position: str | None = Field(None, max_length=100)
    level: EmployeeLevel | None = None
    is_mandatory: bool | None = None
    order: int | None = None
    deadline_days: int | None = Field(None, ge=0, le=365)
    duration_minutes: int | None = Field(None, ge=1, le=480)


class MeetingResponse(MeetingBase):
    """Meeting response schema."""

    id: int
    created_at: datetime
    updated_at: datetime | None = None

    model_config = ConfigDict(from_attributes=True)


class MeetingListResponse(BaseModel):
    """Meeting list response schema."""

    total: int
    meetings: list[MeetingResponse]
    page: int
    size: int
    pages: int

File: admin_web/src/types/meeting.ts
import type { Department } from "./department";

export interface Meeting {
  id: number;
  title: string;
  description: string | null;
  type: string;
  department_id: number | null;
  department: Department | null;
  position: string | null;
  level: string | null;
  deadline_days: number;
  is_mandatory: boolean;
  order: number;
  created_at: string;
  updated_at: string | null;
}

export interface MeetingListResponse {
  total: number;
  meetings: Meeting[];
  page: number;
  size: number;
  pages: number;
}

export interface UserMeeting {
  id: number;
  user_id: number;
  meeting_id: number;
  status: string;
  scheduled_at: string | null;
  completed_at: string | null;
  feedback: string | null;
  rating: number | null;
  created_at: string;
}

export interface UserMeetingListResponse {
  total: number;
  items: UserMeeting[];
  page: number;
  size: number;
  pages: number;
}
